src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/ConcurrentQueueOfTConverter.cs

```diff
@@ -11,6 +12,7 @@ internal sealed class ConcurrentQueueOfTConverter<TCollection, TElement>
     {
         internal override bool CanPopulate => true;
 
+        [DynamicDependency(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor, typeof(ConcurrentQueue<>))]
```

Comment:
Why is this necessary? We try really hard to not use `DynamicDependency` unless it is a last resort.

Reply:
Until now the runtime itself was using `ConcurrentQueue` ctor and so it was never trimmed. 

If we want to keep it friction-less this is one option how.
If we are OK to force users to start doing it themself, we could avoid this DynamicDependency.
Obviously only browser/wasi people will be impacted as thread pool on other platforms will keep it alive anyway.

Reply:
Not a big deal, who needs to deserialize into ConcurrentQueue anyway ?
😉 

- I can disable this test on the browser. 
- Or I can add DynamicDependency in the test, I'm not sure if that would be defeating the test purpose slightly. 

I think it's testing trimming of non-generic collections too, which can keep working.

Please let me know @eerhardt, I don't have strong opinion.

Reply:
I would just work around it in the tests, and not change the product code for it. If you are using the JsonSerializer with trimming/native AOT, the source generator is the only support route.